build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -0,0 +1,1549 @@
+        baseConfig.put("discovery.initial_state_timeout", "0s");
+
+        if (getVersion().getMajor() >= 8) {
+            baseConfig.put("cluster.service.slow_task_logging_threshold", "5s");
+            baseConfig.put("cluster.service.slow_master_task_logging_threshold", "5s");
+        }
+
+        baseConfig.put("action.destructive_requires_name", "false");
+
+        HashSet<String> overriden = new HashSet<>(baseConfig.keySet());
+        overriden.retainAll(settings.keySet());
+        overriden.removeAll(OVERRIDABLE_SETTINGS);
+        if (overriden.isEmpty() == false) {
+            throw new IllegalArgumentException(
+                "Testclusters does not allow the following settings to be changed:" + overriden + " for " + this
+            );
+        }
+        // Make sure no duplicate config keys
+        settings.keySet().stream().filter(OVERRIDABLE_SETTINGS::contains).forEach(baseConfig::remove);
+
+        final Path configFileRoot = configFile.getParent();
+        try {
+            Files.write(
+                configFile,
+                Stream.concat(settings.entrySet().stream(), baseConfig.entrySet().stream())
+                    .map(entry -> entry.getKey() + ": " + entry.getValue())
+                    .collect(Collectors.joining("\n"))
+                    .getBytes(StandardCharsets.UTF_8),
+                StandardOpenOption.TRUNCATE_EXISTING,
+                StandardOpenOption.CREATE
+            );
+
+            final List<Path> configFiles;
+            try (Stream<Path> stream = Files.list(getDistroDir().resolve("config"))) {
+                configFiles = stream.collect(Collectors.toList());
+            }
+            logToProcessStdout("Copying additional config files from distro " + configFiles);
+            for (Path file : configFiles) {
+                Path dest = configFile.getParent().resolve(file.getFileName());
+                if (Files.exists(dest) == false) {
+                    Files.copy(file, dest);
+                }
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException("Could not write config file: " + configFile, e);
+        }
+
+        tweakJvmOptions(configFileRoot);
+        LOGGER.info("Written config file:{} for {}", configFile, this);
+    }
+
+    private void tweakJvmOptions(Path configFileRoot) {
+        LOGGER.info("Tweak jvm options {}.", configFileRoot.resolve("jvm.options"));
+        Path jvmOptions = configFileRoot.resolve("jvm.options");
+        try {
+            String content = new String(Files.readAllBytes(jvmOptions));
+            Map<String, String> expansions = jvmOptionExpansions();
+            for (String origin : expansions.keySet()) {
+                if (content.contains(origin) == false) {
+                    throw new IOException("template property " + origin + " not found in template.");
+                }
+                content = content.replace(origin, expansions.get(origin));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.